src/runtime/vulkan/vulkan_stream.h

```diff
@@ -110,6 +125,7 @@ class VulkanStream {
   std::unordered_map<VkDescriptorSet, std::vector<VulkanStreamToken>> deferred_tokens_;
   std::vector<std::function<void(VulkanStreamState*)>> deferred_kernels_;
   VkCommandPool cmd_pool_;
+  VulkanStreamProfiler* profiler_;
```

Comment:
I think this can live on the stack?

Reply:
> I think this can live on the stack?

The profiler_ is also used from the vulkan_device_api.cc to set/reset the flags used for inserting debug labels into the vulkan stream. Since the profiler_ is used in other files, moving profiler_ into the stack may not work. Please correct if my understanding is wrong

Reply:
I mean, can we just use

`VulkanStreamProfiler profiler_`

and delete new / delete stuff

Reply:
Hi @masahi,
We have implemented the AMD-specific trace capture logic (debug labels insertion) into a separate class AmdRgpProfiler, which is derived from VulkanStreamProfiler. During the VulkanStream creation, the AmdRgpProfiler is instantiated and assigned to VulkanStreamProfiler *profiler. Doing this will help in overriding the default VulkanStreamProfiler->capture() with the AmdRgpProfiler->capture().  If we modify the VulkanStreamProfiler *profiler_ to VulkanStreamProfiler profiler_, the overriding of VulkanStreamProfiler ->capture() with AmdRgpProfiler->capture() will not work. Because of this reason, I would suggest using the existing flow.
Thanks,
Anurag

Reply:
yeah sorry, I missed that ` VulkanStreamProfiler` is a base class.